packages/sdk/src/hooks/useActivities.tsx

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from 'react'
+import { useWalletJsonRecord } from '../providers/WalletJsonStoreProvider'
+import { type ActivityRecord, activityStorage } from '../storage/activities'
+
+export const useActivities = ({ filters }: { filters?: { entityId?: string } } = {}) => {
+  const { record, isLoading } = useWalletJsonRecord<ActivityRecord>(activityStorage.recordId)
+
+  const activities = useMemo(() => {
+    if (!record?.activities) return []
+
+    return [...record.activities]
+      .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
```

Comment:
Maybe a nitpick, but do we need to do this transformation (`new Date(x).getTime()`) every time the sorting function is called? Is it not possible to use `.date` directly?

Reply:
It might be possible, but we store the date internally in the agent and as react state. This means we have to serialize it (currently to an ISO string) and deserialize it.

This means that when we store then we have to call `toIsoString()` and when we fetch it we call `new Date()`. This would slightly improve the sort call, as we do not have to recreate the date for all instances, but the performance gain seems minimal. 

It would be a good cleanup/optimalization, but I would give it a very low prio.